src/client/pages/Offer/Introduction/Sidebar/StartDate.tsx

```diff
@@ -307,18 +301,25 @@ export const StartDate = ({
 }: StartDateProps) => {
   const textKeys = useTextKeys()
   const [showError, setShowError] = useState(false)
-  const [loadingQuoteIds, setLoadingQuoteIds] = React.useState<Array<string>>(
-    [],
-  )
+  const [loadingQuoteIds, setLoadingQuoteIds] = useState<Array<string>>([])
 
   const handleSelectNewStartDate = async (
     newDateValue: Date | null,
     quoteIds: Array<string>,
   ) => {
     try {
       setShowError(false)
-      setLoadingQuoteIds(quoteIds)
-      await onSelect(newDateValue, quoteIds)
+
+      const quotesToBeUpdated = [...quoteIds]
```

Comment:
Hm... I looked at the PR that introduced line 490 but I don't understand the reason why that Car-specific thing is there.

The logic goes like this:

> In `createQuoteBundle`, while trying to create the quotes to send to BE, we try to determine the start date for each quote. If `quoteIds` (parameter to `onSelect`, I guess it's the selected quotes) contains `id` then we should use the new date, otherwise use the old date.

And when Car was introduced the logic also applies to:

> If `quote::id` is Car and Car is selected then we should use the new date, otherwise use the old date.

It's not clear to me why the old if statement didn't suffice, which is a nice reminder to document _why_ we do things, not _what_ they do 🙃 

Oh... as I was writing this maybe I have an idea! 

When not showing a car quote we can (and want to perhaps) set individual start dates for different insurances. House and Accident should perhaps start at different dates. 

But for car, if you select Traffic insurance and then set a start date, and _then_ change to Half coverage the recently changed start date will revert back to the default without the user realising. So I think we did this change so that the date set by the user stays.. 

So the logic goes:

> In `createQuoteBundle`, while trying to create the quotes to send to BE, we try to determine the start date for each quote. If `quoteIds` contains `id` then we should use the new date, otherwise use the old date. AND if we're displaying Car, update all (three) quotes to the newly selected date.


(and actually the ticket for the PR says the reason as well: https://hedvig.atlassian.net/browse/GRW-1215 😂 )

Reply:
So, to sum it up: I don't think you can get around the issue, since Car quotes behave quite different from all other types.

But perhaps we could also argue that, if you have an Offer page with two bundles: 
* A: Home+accident
* B: Home

Then if you would have B selected and set start date 2022-12-01 for Home and then switch to A, that date should still be set for Home. So perhaps this could be generalized to say that we always want to edit the date for the _type_ of quote that you currently have selected.

But then again, that case might be handled by BE given quotes and possible variations, which is not the case for Car... Because in the case above (A/B), I guess you only have two actual quotes, Home and Accident, but they appear in two possible variations as a bundle? Or?

Reply:
That's a valid point. By editing the quote instead of recreating them I believe we'll be covering the A/B scenario that you've mentioned. But that still will not be enough for car, I'm afraid. As you've pointed out it's different from other types. We create 3 different and they need to _share_ the same start date. Using `edit_quote` mutation or not we'll need to make that change manually.

Thanks for the explanation. It really helped :)